dlt/destinations/sql_client.py

```diff
@@ -292,11 +302,26 @@ def _truncate_table_sql(self, qualified_table_name: str) -> str:
             return f"DELETE FROM {qualified_table_name} WHERE 1=1;"
 
 
+class WithSqlClient(JobClientBase):
+    @property
+    @abstractmethod
+    def sql_client(self) -> SqlClientBase[TNativeConn]: ...
+
+    def __enter__(self) -> "WithSqlClient":
+        return self
+
+    def __exit__(
+        self, exc_type: Type[BaseException], exc_val: BaseException, exc_tb: TracebackType
+    ) -> None:
+        pass
+
+
 class DBApiCursorImpl(DBApiCursor):
     """A DBApi Cursor wrapper with dataframes reading functionality"""
 
     def __init__(self, curr: DBApiCursor) -> None:
         self.native_cursor = curr
+        self.columns = {}
```

Comment:
where columns are coming from? they look they are always empty.
1. ideally we know the columns in a advance (from table or via sqlglot - which I doubt will work)
2. or we set the columns after first row/chunk is available by reflecting back from data: we have code for python object (in Schema - infer column, in arrow obviously, and we lack similar things for pandas but that's trivial to do)

Reply:
In case where we do not know the schema (sql queries) they are set to this:
`{c: {"name": c, "nullable": True} for c in self._get_columns()}`

In case where we know the schema they are set to the tablecolumns of the table we are querying.